src/main/java/pl/cleankod/exchange/entrypoint/AccountController.java

```diff
@@ -1,59 +1,63 @@
+    public AccountController(final FindAccountUseCase findAccountUseCase, final JsonWriter writer) {
         this.findAccountUseCase = findAccountUseCase;
+        this.writer = writer;
     }
 
     @GetMapping(path = "/{id}")
-    public ResponseEntity<Account> findAccountById(@PathVariable String id, @RequestParam(required = false) String currency) {
-        return Optional.ofNullable(currency)
-                .map(s ->
-                        findAccountAndConvertCurrencyUseCase.execute(Account.Id.of(id), Currency.getInstance(s))
-                                .map(ResponseEntity::ok)
-                                .orElse(ResponseEntity.notFound().build())
-                )
-                .orElseGet(() ->
-                        findAccountUseCase.execute(Account.Id.of(id))
-                                .map(ResponseEntity::ok)
-                                .orElse(ResponseEntity.notFound().build())
-                );
+    @ApiOperation(value = "Retrieve account details with account id")
+    @ApiResponses({
+        @ApiResponse(code = 200, message = "Ok"),
+        @ApiResponse(code = 404, message = "Account does not exist", response = ErrorResponse.class)
+    })
+    public ResponseEntity<AccountDto> findAccountById(@PathVariable String id, @RequestParam(required = false) String currency) {
+        final var account = findAccountUseCase.execute(Account.Id.of(id), currency);
+        return ResponseEntity.ok(account.toDto());
     }
 
     @GetMapping(path = "/number={number}")
-    public ResponseEntity<Account> findAccountByNumber(@PathVariable String number, @RequestParam(required = false) String currency) {
+    @ApiOperation(value = "Retrieve account details with account number")
+    @ApiResponses({
+        @ApiResponse(code = 200, message = "Ok", response = AccountDto.class),
+        @ApiResponse(code = 404, message = "Account does not exist", response = FindAccountFailedReason.class)
+    })
+    public Object findAccountByNumber(@PathVariable String number, @RequestParam(required = false) String currency, HttpServletResponse resp) {
         Account.Number accountNumber = Account.Number.of(URLDecoder.decode(number, StandardCharsets.UTF_8));
-        return Optional.ofNullable(currency)
-                .map(s ->
-                        findAccountAndConvertCurrencyUseCase.execute(accountNumber, Currency.getInstance(s))
-                                .map(ResponseEntity::ok)
-                                .orElse(ResponseEntity.notFound().build())
-                )
-                .orElseGet(() ->
-                        findAccountUseCase.execute(accountNumber)
-                                .map(ResponseEntity::ok)
-                                .orElse(ResponseEntity.notFound().build())
-                );
+        return findAccountUseCase.execute(accountNumber, currency)
+            .fold(account -> writer.mapObjectToString(account.toDto()),
```

Comment:
Quite a lot of manual operations happening here. Any ideas on how to improve that?